### PR TITLE
feat(kad): impl Display on QueryId

### LINKED ISSUE #4813: kad: impl Display on QueryId
### Description

Currently, only `Debug` is implemented on `protocols::kad::src::query.rs::QueryId`. It would be great to also add a `Display` implementation for logging purposes, for example.

### Motivation

For logging purposes, we would like to get (using `tracing`) something like `query_id=3` instead of what is currently done `query_id=QueryId(3)`. 

Indeed, since only `Debug` implemented, we currently have to do `debug!(?query_id, "Got a query")` which render `query_id=QueryId(3)`.

We would like to be able to do `debug!(%query_id, "Got a query")` to get `query_id=3` which would be shorter and cleaner.

### Requirements

1. implement `std::fmt::Display` on `protocols::kad::src::query.rs::QueryId`: `format!("{}", QueryId(3))` should return `"3"`.

### Open questions

_No response_

### Are you planning to do it yourself in a pull request ?

Yes

### PR DESCRIPTION
**Pull request by stormshield-frb**
## Description

Fixes: #4813.

## Notes & open questions

N/A

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/kad/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.45.1 - unreleased
+
+- Add `std::fmt::Display` implementation on `QueryId`
```
Comment (dariusc93) [inline protocols/kad/CHANGELOG.md]:
```suggestion
- Add `std::fmt::Display` implementation on `QueryId`.
```

File: protocols/kad/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+  See [PR 4814](https://github.com/libp2p/rust-libp2p/pull/4814)
```
Comment (dariusc93) [inline protocols/kad/CHANGELOG.md]:
```suggestion
  See [PR 4814](https://github.com/libp2p/rust-libp2p/pull/4814).
```

File: protocols/kad/CHANGELOG.md (hunk 1 above)
Comment (stormshield-frb) [reply protocols/kad/CHANGELOG.md]: Done

File: protocols/kad/CHANGELOG.md (hunk 2 above)
Comment (stormshield-frb) [reply protocols/kad/CHANGELOG.md]: Done

File: protocols/kad/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 0.45.1 - unreleased
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md:1]: You'll also need to bump the version in the manifest please.

**Review by thomaseizinger — COMMENTED** [review]
Thanks, I think this makes sense. I kicked CI!

File: protocols/kad/CHANGELOG.md (hunk 3 above)
Comment (thomaseizinger) [reply protocols/kad/CHANGELOG.md:1]: Also in the root manifest.

Comment (stormshield-frb) [reply protocols/kad/CHANGELOG.md:1]: Version bumped in `protocols/kad/Cargo.toml` and `Cargo.toml`. I'll let you resolve this conversation if it was what you meant.

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**